protocol/contracts/v2/TreasuryReservesVault.sol

```diff
@@ -4,73 +4,88 @@ pragma solidity ^0.8.17;
 
 import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
 
 import { ITreasuryReservesVault } from "contracts/interfaces/v2/ITreasuryReservesVault.sol";
 import { ITreasuryPriceIndexOracle } from "contracts/interfaces/v2/ITreasuryPriceIndexOracle.sol";
-import { ITempleERC20Token } from "contracts/interfaces/core/ITempleERC20Token.sol";
 import { ITempleStrategy, ITempleBaseStrategy } from "contracts/interfaces/v2/strategies/ITempleBaseStrategy.sol";
 import { ITempleDebtToken } from "contracts/interfaces/v2/ITempleDebtToken.sol";
 
 import { CommonEventsAndErrors } from "contracts/common/CommonEventsAndErrors.sol";
 import { TempleElevatedAccess } from "contracts/v2/access/TempleElevatedAccess.sol";
 
+/**
+ * @title Treasury Reserves Vault (TRV)
+ *
+ * @notice Temple has various strategies which utilise the treasury funds to generate 
+ * gains for token holders.
+ * 
+ * The maximum amount of funds allocated to each strategy is determined by governance, 
+ * and then each strategy can borrow/repay as required (up to the cap).
+ * 
+ * When strategies borrow funds, they are issued `dToken`, an accruing debt token representing
+ * the debt to the temple treasury. This is used to compare strategies performance, where
+ * we can determine an equity value (assets - debt).
+ *
+ *    Strategies can borrow different types of tokens from the TRV, and are minted equivalent internal debt tokens eg:
+ *      DAI => minted dUSD
+ *      TEMPLE => minted dTEMPLE
+ *      ETH => minted dETH
+ *   
+ *   Each of the dTokens are compounding at different risk free rates, eg:
+ *      dUSD: At DAIs Savings Rate (DSR)
+ *      dTEMPLE: 0% interest (no opportunity cost)
+ *      dETH: ~avg LST rate
+ *   
+ *   And so each token which can be borrowed has separate config on how to pull/deposit idle funds.
+ *   For example, this may be:
+ *      DAI => DSR base strategy
+ *      TEMPLE => direct Temple mint/burn 
+ *      ETH => just hold in a vault
```

Comment:
Staking yes, unstaking there's generally a wait period I think. Yep we'd have to think about what the base strategy does in this case - it might just hold it, or maybe there is some other (extremely low risk) LSD thing we can use.

The dETH interest rate would be at that average rate regardless, because it represents the opportunity cost of it going into a strategy rather than being staked.